### PR TITLE
Simplify example to not use `translate` requirement

### PR DESCRIPTION
**Pull request by Eric-Arellano**
As explained in https://github.com/pantsbuild/example-python/pull/84, we think it's valuable for this example repository to be simple. In particular, we are demoing Pants itself, not the underlying code. 

This simplifies the project to use our own hardcoded `translate.json`, rather than the much more robust `translate` requirement. We still demonstrate the same things `translate` was showing for us:

* third-party management
* loading resource files

There's an added benefit that the repository is now much faster to build, especially on Apple Silicon where `lxml` must be compiled from source. Likewise, I no longer have to force Pants to use Py39+ for this to build on my M1! The default of 3.7+ now works.

### CODE REVIEW

File: helloworld/greet/BUILD [hunk 1]
```diff
@@ -3,13 +3,19 @@
...
+)
 
 # This target sets the metadata for all the Python test files in this directory.
-#
-# * `sources` defaults to ['*_test.py', 'test_*.py', 'conftest.py'].
-# * `dependencies` are inferred.
-python_tests(name='tests')
+python_tests(
+    name="tests",
+)
```
Comment (Eric-Arellano) [inline helloworld/greet/BUILD:15]: I very much prefer the style of keeping it on one-line. But `tailor` generates it this way, so probably good to be consistent with that.

File: helloworld/greet/BUILD [hunk 2]
```diff
@@ -3,13 +3,19 @@
...
+
+# This target teaches Pants about our JSON file, which allows for other targets to depend on it.
```
Comment (stuhood) [inline helloworld/greet/BUILD:17]:
```suggestion
# This target teaches Pants about our JSON file, which allows other targets to depend on it.
```

**Review by Eric-Arellano — COMMENTED** [review]
Note that because I deleted `config.py` and `config_test.py`, we no longer have an example of >1 test per directory...I'm not sure how much we care about that.

Preserving `config.py` was too complex imo. We already have an example of loading a `resource` file, I really don't think we want to demonstrate that >1 time.

File: helloworld/greet/BUILD [hunk 3]
```diff
@@ -3,13 +3,19 @@
...
+resources(
+    name="translations",
+    sources=["translations.json"],
```
Comment (stuhood) [inline helloworld/greet/BUILD:20]: Maybe glob `*.json`? A bit shorter, and less fragile.

**Review by stuhood — APPROVED** [review]

File: helloworld/greet/BUILD (hunk 3 above)
Comment (Eric-Arellano) [reply helloworld/greet/BUILD:20]: In 2.8, I will update this to use `resource` instead of `resources`

**Review by benjyw — APPROVED** [review]

**Pull request merged**